File: 1212-sequential-digits/sequential-digits.cpp
class Solution {
public:
    vector<int> sequentialDigits(int low, int high) {
        vector<int> v;

        // Base string containing sequential digits
        string str = "123456789";

        // Get the number of digits in low and high
        string l = to_string(low);
        string h = to_string(high);

        // Generate numbers of every possible length
        for (int i = l.length(); i <= h.length(); i++) {

            // Slide a window of length i
            for (int j = 0; j <= 9 - i; j++) {

                // Extract i consecutive digits
                int num = stoi(str.substr(j, i));

                // Check if it lies in the required range
                if (num >= low && num <= high) {
                    v.push_back(num);
                }
            }
        }

        return v;
    }
};

File: 1212-sequential-digits/sequential-digits.java
class Solution {
    public List<Integer> sequentialDigits(int low, int high) {
        List<Integer> list=new ArrayList<>();
        String str="123456789";
        String l=low+"";
        String h=high+"";
        for(int i=l.length(); i<=h.length(); i++){
            for(int j=0; j<=9-i; j++){
                int num=Integer.parseInt(str.substring(j,j+i));
                if(num<=high && num>=low){
                    list.add(num);
                }
            }
        }
        return list;
        
    }
}